### DISCUSSION TITLE
Limit file size to fixed kb

### DISCUSSION THREAD

**Discussion by JanCourtheyn** [ImageMagick/ImageMagick#3628] [Help] [answered]
Dear all,

For web usage, I'm trying to limit certain images between 20 and 100kb.
I tried the following commands already:
-gaussian-blur 0.05 
-define jpeg:extent=20kb
-quality 50

But none of these really decrease the file size compared if I don't use these parameters.
- Without rules: 287kb
- With extent:20kb = 197kb (but very bad image quality)
- with quality 50 = 224kb
- gaussian = 273kb

Does anybody know a good solution without losing too much quality?
This is the default rule that is used:

magick.exe convert "INPUT.PSD"[4] -profile sRGB.icc -resize 280x280 -gravity center -extent 300x300 -density 72x72 "WEBIMAGE.JPG"

Thanks in advance.

Kind regards,
Jan

**Comment by snibgo** [comment]
What version of IM, on what platform?

"-define jpeg:extent=20kb" should ensure the output jpeg is less than 20 KB, by adjusting the quality setting. Don't use this with a "-quality" setting.

If that doesn't work, and you are using the latest IM version, please link to your input PSD.

**Reply by JanCourtheyn** [reply]
For your information:
- the file: https://drive.google.com/file/d/1LV-grzeeTL1ShL3P9efj4gZNaW84lFn9/view?usp=sharing
- Version I'm using: ImageMagick-7.0.11-Q16-HDRI (on local machine for testing)

**Comment by snibgo** [comment, marked as answer]
That image contains a large amount of metadata, which messes up the IM size calculations. I suggest you convert to sRGB, and "-strip" the non-image data:
```
magick 46631035.psd[4] -profile sRGB.icc -strip -define jpeg:extent=20kb x.jpg
```
x.jpg is 22988 bytes. I don't know why this is (slightly) larger than 20kb.

**Reply by JanCourtheyn** [reply]
Thanks a lot, that -strip did indeed a lot of improvement to the file size.
